Lib/socket.py

```diff
@@ -628,28 +628,81 @@ def socketpair(family=AF_INET, type=SOCK_STREAM, proto=0):
         if proto != 0:
             raise ValueError("Only protocol zero is supported")
 
-        # We create a connected TCP socket. Note the trick with
-        # setblocking(False) that prevents us from having to create a thread.
-        lsock = socket(family, type, proto)
-        try:
-            lsock.bind((host, 0))
-            lsock.listen()
-            # On IPv6, ignore flow_info and scope_id
-            addr, port = lsock.getsockname()[:2]
-            csock = socket(family, type, proto)
+        import secrets  # Delay import until actually needed.
+        auth_len = secrets.DEFAULT_ENTROPY
+        reason = "unknown"
+        for _ in range(5):
+            # We do not try forever, that'd just provide another process
+            # the ability to make us waste CPU retrying.  In all normal
+            # circumstances the first connection auth attempt succeeds.
+            s_auth = secrets.token_bytes()
+            c_auth = secrets.token_bytes()
```

Comment:
This might seem silly, but how about deferring the generation of the tokens until the sockets are authenticated?

Reply:
This review is outdated given new method of authenticating.